osquery/distributed/distributed.cpp

```diff
@@ -117,11 +119,56 @@ Status Distributed::serializeResults(std::string& json) {
     doc.add(result.request.id, arr, queries_obj);
     doc.add(result.request.id, result.status.getCode(), statuses_obj);
     doc.add(result.request.id, result.message, messages_obj);
+
+    RecursiveLock lock(performance_mutex_);
+    auto obj = doc.getObject();
+    if (performance_.count(result.request.id) > 0) {
+      auto perf = performance_[result.request.id];
+      obj.AddMember("executions",
+                    static_cast<uint64_t>(perf.executions),
+                    obj.GetAllocator());
+      obj.AddMember("last_executed",
+                    static_cast<uint64_t>(perf.last_executed),
+                    obj.GetAllocator());
+      obj.AddMember("output_size",
+                    static_cast<uint64_t>(perf.output_size),
+                    obj.GetAllocator());
+      obj.AddMember("wall_time",
+                    static_cast<uint64_t>(perf.wall_time),
+                    obj.GetAllocator());
+      obj.AddMember("wall_time_ms",
+                    static_cast<uint64_t>(perf.wall_time_ms),
+                    obj.GetAllocator());
+      obj.AddMember("last_wall_time_ms",
+                    static_cast<uint64_t>(perf.last_wall_time_ms),
+                    obj.GetAllocator());
```

Comment:
I was going to ask if the `last_*` and `*` metrics are both necessary in the context of distributed queries, but on further reflection I think it could be useful. In Fleet, for example, some queries are sent repeatedly (such as to populate host details), while others are sent only once. For the repeated queries it could be useful to have the aggregated metrics rather than just the last execution metrics.

Reply:
If they use the same ID then they probably make sense to send the non `last_*` metrics. If they don't then it would make more sense to completely get rid of the `performance_` object and just create a new map on every execution, which would somewhat simplify the code for this.